### ISSUE TITLE
FM-3 — Foundation-model registry: canonical naming, deprecation mapping, v1 loaders, and the nightly badge matrix

### ISSUE THREAD

**Issue by aacostadiaz** [ACEsuit/mace#1588]
**Depends on:** CORE-2 (#1556), FM-2 (#1587), FM-4 (#1636), DEP-1 (#1583) · **Blocks:** —

**Prerequisite:** the Phase 0 characterization suite is committed — goldens, fixtures and the single-source tolerance table under `tests/golden/`, characterization tests under `tests/unit/`. References of the form `P0-N` name that work.

**Context:** Today the "model registry" is three hardcoded URL dicts in `mace/calculators/foundations_models.py:92-130` — `mace_mp_urls` (16 aliases: `small`/`medium`/`large`, the `*-0b`/`*-0b2`/`*-0b3` generations, `medium-mpa-0`, `small-omat-0`/`medium-omat-0`, `mace-matpes-pbe-0`/`mace-matpes-r2scan-0`, `mh-0`/`mh-1`), `mace_off_urls` (OFF23 small/medium/large) and `polar_model_urls` (`polar-1-s/m/l`) — plus `mace_mdp_default_url` (`:128-130`), a one-entry `urls` dict inlined in `mace_omol` (`:541-543`, key `extra_large`), and an inline ANI-CC URL (`:491`). That is **23 named aliases across five sources**, three of them not module-level, and there is no single place that enumerates them. The naming is inconsistent across families (`medium-0b3` vs `mace-matpes-pbe-0` vs `mh-0` vs `extra_large`), there is no deprecation signal of any kind, DOIs and citations live only in docstrings (`:255-262`, `:389`, `:475`, `:528`), and license handling is three unrelated ad-hoc sites — an ASL URL set with a printed notice in `download_mace_mp_checkpoint` (`:157-166`), a separate always-printed ASL notice inside `mace_off` (`:417-422`), and a third inside `mace_omol` (`:566-570`). The `mace_mp()` default is `medium-mpa-0`, and it is **silent in the common case**: when the bundled copy is present the function returns at `:144-145` with no log line at all, and only the download path prints the default-selection notice (`:153-156`). FM-2 (#1587) has converted and published every artifact on the migration list as a neutral safetensors+JSON checkpoint; FM-4 (#1636) owns resolving and fetching them. This ticket builds the registry, naming discipline, user-facing loaders and support-statement machinery on top.

**Interface & constraints:**

1. **Canonical naming scheme:** `<label>-[data-]<hyperparam>` (e.g. `mace-mpa-0-medium`; the hyperparam part is the size tier — small/medium/large — and the data part names the training set / generation, as in `mh-0` vs `mh-1`). Every registry entry has exactly one canonical id in this scheme.
2. **Deprecation mapping — total, never silent.** Every alias accepted by the legacy loaders maps to either a canonical id or an explicit tombstone: all 16 `mace_mp_urls` keys, the 3 `mace_off_urls` keys, the 3 `polar_model_urls` keys, `mace_omol`'s `extra_large`, and the four `None` defaults (`mace_mp` → `medium-mpa-0`, `mace_off` → `medium`, `mace_omol` → `extra_large`, `mace_mdp` → its single URL). Today none of these emits any deprecation signal; v1 resolves a deprecated alias successfully but emits a runtime warning naming the canonical replacement. Registry entries carry a status (**recommended** vs **deprecated**, with "recommended for which configurations and chemistries" as structured text), and the docs/website surface the same labels. File/shortcut renaming to force user attention is an allowed mechanism for hard deprecations.
3. **Registry records.** One record per published model: canonical id, deprecated aliases, artifact reference (the FM-2 (#1587) neutral artifact; resolved to URL + DOI by FM-4 (#1636)), DOI, structured citations, license, status, network hyperparameters, and the badge reference. License is a **field**, not three code sites: an ASL entry reproduces the acceptance notice legacy prints, an MIT entry prints nothing, and the notice text lives once. Metadata rides the CORE-2 (#1556) `ModelMetadata` schema (resolved config, code version + git commit, E0 details, DOI, citations) — no second metadata format. The registry table and its query API live in `mace_core` (torch-free) — mace-jax consumes canonical ids and artifact refs for its own `SUPPORTED_MODELS` without importing torch; the torch loaders live in `mace_torch.calculators.foundations`.
4. **Query/extension API:** report network hyperparameters for any entry; extend a loaded model by adding species and adding readouts (coordinate the species path with FT-3 (#1582)'s new-species parameter-initialization spec rather than inventing a second one).
5. **Loaders — six families, not two.** v1 ships the equivalents of `mace_mp`, `mace_off`, `mace_omol`, `mace_mdp` and `mace_polar` (`mace_anicc` is dropped, §7), each returning the DEP-1 (#1583) ASE calculator. Preserved surface, per family:

| loader | default model | default dtype | model_type driven into the calculator |
|---|---|---|---|
| `mace_mp` | `medium-mpa-0` | `float32` | `MACE` |
| `mace_off` | `medium` | `float64` | `MACE` |
| `mace_omol` | `extra_large` | `float64` | `MACE`, `head="omol"` forced |
| `mace_mdp` | its single artifact | `float64` | `DipolePolarizabilityMACE` (the checkpoint is an `AtomicDielectricMACE`); warns that the model is dipole/polarizability-only and rejects any other `model_type` |
| `mace_polar` | **none — `model` is required** | `float32` | `PolarMACE` (row registered by ELEC-2 (#1592)) |

   The per-family dtype defaults are behaviour, not style: `mace_mp` defaulting to fp32 and `mace_off` to fp64 is what today's users get, and a uniform default would silently change every downstream result. Shared kwargs `model` / `device` / `default_dtype` / `return_raw_model` are preserved across all five, along with `device=""` meaning "cuda if available else cpu". The `mace_mp` D3-dispersion option keeps its full current surface — `dispersion`, `dispersion_damping`, `dispersion_xc`, `dispersion_cutoff`, `dispersion_coord_cutoff` — **including the `damping` alias**: `damping` was the released name, it still arrives through `**kwargs`, and because `**kwargs` are also forwarded to `TorchDFTD3Calculator` an unhandled alias raises a multiple-values `TypeError` instead of being honoured. All three behaviours are already pinned by `tests/unit/test_foundations_models.py` (forwarding, the legacy alias, the `bj` default); port those tests, do not re-derive the contract. Calling a loader with `model` omitted resolves to that family's **documented default canonical id** and logs the id it selected — closing today's silent-bundled-default path. That default is a field on the registry, not a constant in the loader. Loaders load **FM-2 (#1587) neutral artifacts only** — a legacy pickle path is rejected with a message pointing at the FM-1 (#1586) migration tool. Model resolution, download and caching are FM-4 (#1636)'s; the loaders call it.
6. **Validation badge matrix.** A nightly job emits one row per registry entry: model name, code version, expected numerical values (checked against the committed FM-2 (#1587)/P0-2/P0-3a goldens, never against a fresh run's self-consistency), and the hardware infrastructure used for validation. It hangs off the existing `nightly.yaml` `foundations` job's shape (`network`-marked, `MACE_CI_ALLOW_NETWORK=1`, `~/.cache/mace` keyed on the registry file) rather than inventing a second download-bearing workflow. The badge **is** the support statement: "supported" is a four-gate property (architecture closure, published artifact + committed golden, conformance green, registry row + badge), and this matrix is the fourth gate — for torch and, for the three JAX-supported artifacts (MP-0 small, MPA-0 medium, OFF23 medium), for JAX. Fine-tuned descendants of registry models are "expected to work, not supported" — they get no row.
7. **Roster and tombstones** (view of the settled migration list): MACE-MP aliases incl. `mh-0`/`mh-1` (heads intact), MACE-OFF, MACE-MDP, MACE-OMOL and MACE-Polar — all converted by FM-2 (#1587) (the polar artifacts onto the model ELEC-2 (#1592) registers in Phase 3); MACE-ANI-CC is **dropped** — it registers as a tombstone resolving to MACE-OFF, never as a loadable entry, and `mace_anicc` has no v1 successor. Multi-head entries register as multi-head; a single-head export is `mace model select-head`, a separate step, not a registry variant.

**Explicit DROP rows** (each is behaviour v1 does not carry, with its reason):

| Dropped | Reason |
|---|---|
| `mace_anicc` and the MACE-ANI-CC artifact | superseded by MACE-OFF for organic chemistry; not converted by FM-2 (#1587), so there is nothing for a loader to load. Also removes a 35 MB pickle from the source tree |
| `--downloads_dir` | already a **no-op** in develop: defined at `mace/tools/arg_parser.py:60`, defaulted to `<work_dir>/downloads` at `mace/tools/arg_parser_tools.py:25-26`, and read by nothing. Dropping it changes no behaviour |
| `mace_anicc`'s literal `device="cuda"` default | the only loader without the "cuda if available else cpu" fallback; it dies on a CPU-only host. Dies with the loader |
| the three per-site ASL print statements | replaced by one license field on the registry record, printing the same text |

**Two constraints on this ticket:** (a) The **DOI repository choice is an external dependency with no code owner yet** (not necessarily Zenodo, for space reasons); the registry schema must not hard-code a repository, and the DOI-facing acceptance items below cannot close until a named human owner settles it. (b) If Phase 4 capacity runs out, the badge matrix is the designated cut *inside this ticket* — cutting it is an explicit, recorded deviation, never a quiet omission.

**Task:**
1. `mace_core`: registry record schema (on top of CORE-2 (#1556) `ModelMetadata`), the canonical-naming rule, the alias→canonical deprecation table covering all 23 legacy aliases plus the four `None` defaults, and the query API (hyperparams, DOI, citations, license, status).
2. `mace_torch.calculators.foundations`: the five v1 loaders per the table in §5 — alias resolution with deprecation warnings, per-family defaults, DEP-1 (#1583) calculator construction, `return_raw_model`, `device`/`default_dtype`, the D3-dispersion option with its ported tests, license notices driven from the registry field.
3. The extension API: report-hyperparameters, add-species (per FT-3 (#1582)'s init spec), add-readouts; each with a round-trip test on a small registry model artifact.
4. The nightly badge-matrix job: iterate the registry, evaluate each model against its committed goldens, emit the per-model badge rows (name, code version, expected values, hardware) as a nightly artifact consumable by the docs/website.
5. Wire the deprecated-alias warnings and recommended/deprecated labels into the docs export so website and runtime say the same thing.

**Out of scope:** converting or publishing artifacts (FM-2 (#1587)); the end-user migration/conversion tool (FM-1 (#1586)); artifact resolution, download, caching and the bundled-artifact decision (FM-4 (#1636)); MACE-Polar conversion and its registry row's artifact (ELEC-2 (#1592)); the JAX `SUPPORTED_MODELS` table itself (JAX-1 (#1589) — it consumes this registry's ids/refs); the migration guide (REL-1 (#1603)); single-head export (`mace model select-head`).

**Acceptance criteria:**
- [ ] All 23 legacy aliases (16 MP + 3 OFF + 3 Polar + `extra_large`) and the four `None` defaults resolve through the deprecation table to a canonical `<label>-[data-]<hyperparam>` id or an explicit tombstone; resolving a deprecated alias emits a warning naming the replacement; no alias resolves silently to something else.
- [ ] The registry query API returns hyperparameters, DOI, structured citations, license, and recommended/deprecated status for every entry; the ASL notice text exists once and is emitted from the license field for exactly the entries that carry it.
- [ ] The five v1 loaders return a working DEP-1 (#1583) calculator, each with its documented default model and default dtype from the §5 table; `return_raw_model`, `device` (including `device=""` auto-selection), `default_dtype` honoured; `mace_polar` still requires an explicit model; `mace_mdp` warns and rejects a non-`DipolePolarizabilityMACE` `model_type`; `mace_omol` forces `head="omol"`.
- [ ] The D3-dispersion contract is preserved: `dispersion_damping` forwarded, the legacy `damping` alias honoured without a multiple-values `TypeError`, `bj` as default, plus `dispersion_xc`/`dispersion_cutoff`/`dispersion_coord_cutoff`.
- [ ] Loaders accept only FM-2 (#1587) neutral artifacts; a legacy pickle is rejected with a message naming the FM-1 (#1586) migration tool.
- [ ] Add-species and add-readouts round-trip on a registry model; reported hyperparameters match the artifact's resolved config.
- [ ] The nightly badge matrix emits one row per registry entry (model name, code version, expected numerical values vs committed goldens, hardware), and a golden mismatch turns the row red (badge = support statement).
- [ ] Registry data and query API are importable without torch (asserted).
- [ ] A loader called with no `model` argument resolves to the registry's declared default for that family and logs the canonical id it selected — no path returns a default without saying so.
- [ ] Every DROP row above is recorded in the registry or the migration guide, none by omission.

**Verify:**
```bash
python -m pytest packages/mace-core/tests/test_model_registry.py -v          # naming, all 23 aliases, deprecation warnings, torch-free import
python -m pytest packages/mace-torch/tests/foundations/ -m "not network" -v  # five loaders on local/cached artifacts, dispersion, extension API
MACE_CI_ALLOW_NETWORK=1 python -m pytest packages/mace-torch/tests/foundations/ -m network -v  # opt-in: real resolution against published artifacts
```
**Review focus:** the alias→canonical mapping (completeness against all five legacy sources — a dropped alias is a silent user-facing break, and three of those sources are inline in loader bodies where they are easy to miss), the per-family dtype defaults (a uniform default silently changes every result), the naming table itself (not a solo choice), and that the badge matrix gates on committed goldens rather than self-consistent fresh runs.

---
